exchange/exchange.go

```diff
@@ -629,6 +641,7 @@ func (e *exchange) getAllBids(
 			reqInfo := adapters.NewExtraRequestInfo(conversions)
 			reqInfo.PbsEntryPoint = bidderRequest.BidderLabels.RType
 			reqInfo.GlobalPrivacyControlHeader = globalPrivacyControlHeader
+			reqInfo.BidderRequestStartTime = bidderRequestStartTime
```

Comment:
At any given time, if the system doesn't have enough resources for a parallel run, it may run the goroutines one after the other in a sequential fashion which opens the possibility of `bidder.me.RecordOverheadTime(metrics.PreBidder, time.Since(pbsRequestStartTime))` logging more time for the latter elements of `bidderRequests` than it should. Take the case where `bidderRequests` comes with two elements and, because of low system resources, both `go bidderRunner()` function calls run sequentially: the first time we call `bidder.me.RecordOverheadTime(metrics.PreBidder, time.Since(pbsRequestStartTime))` in line 522 of `exchange/bidder.go`, we would accurately log `go bidderRunner()`'s first runtime because at this point in time `pbsRequestStartTime` only contains the time it took for this first `go bidderRunner()` run. But, by the time we make the second `bidder.me.RecordOverheadTime(metrics.PreBidder, time.Since(pbsRequestStartTime))` call in the second `go bidderRunner()` run, `pbsRequestStartTime` contains the time it took for both the first **and the second** `go bidderRunner()` run. Does this make sense? 

In code, what I'm trying to say is that `bidder.me.RecordOverheadTime(metrics.PreBidder, time.Since(pbsRequestStartTime))` should log the `start` time that is instantiated inside the `go bidderRunner()` goroutine because in doing so we always ensure that `bidder.me.RecordOverheadTime(metrics.PreBidder, time.Since(pbsRequestStartTime))` records the time it took since the beginning of the current `go bidderRunner()` run:
```
599   func (e *exchange) getAllBids(
600       ctx context.Context,
601 *-- 12 lines: bidderRequests []BidderRequest,----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
613       *openrtb_ext.Fledge,
614       bool) {
615       // Set up pointers to the bid results
616       adapterBids := make(map[openrtb_ext.BidderName]*entities.PbsOrtbSeatBid, len(bidderRequests))
617       adapterExtra := make(map[openrtb_ext.BidderName]*seatResponseExtra, len(bidderRequests))
618       chBids := make(chan *bidResponseWrapper, len(bidderRequests))
619       bidsFound := false
620   
621       e.me.RecordOverheadTime(metrics.MakeBidderRequests, time.Since(pbsRequestStartTime))
622   
623 -     bidderRequestStartTime := time.Now()
624       for _, bidder := range bidderRequests {
625           // Here we actually call the adapters and collect the bids.
626           bidderRunner := e.recoverSafely(bidderRequests, func(bidderRequest BidderRequest, conversions currency.Conversions) {
627               // Passing in aName so a doesn't change out from under the go routine
628 *-- 11 lines: if bidderRequest.BidderLabels.Adapter == "" {--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
639               start := time.Now()
640   
641               reqInfo := adapters.NewExtraRequestInfo(conversions)
642               reqInfo.PbsEntryPoint = bidderRequest.BidderLabels.RType
643               reqInfo.GlobalPrivacyControlHeader = globalPrivacyControlHeader
644 -             reqInfo.BidderRequestStartTime = bidderRequestStartTime
    +             reqInfo.BidderRequestStartTime = start
645   
646 *-- 41 lines: bidReqOptions := bidRequestOptions{------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
687           }, chBids)
688           go bidderRunner(bidder, conversions)
689       }
exchange/exchange.go
```

Reply:
changes recommended makes sense. 

https://github.com/prebid/prebid-server/pull/2658/commits/49594b62b40632065dc08a4cb0407a4d445b8068 makes change to use `start` instead of `bidderRequestStartTime`